windows/10.x.x/sitecore-custom-mssql/sitecore-custom-modules-mssql-init/xp-build.json

```diff
@@ -0,0 +1,62 @@
+{
+  "tags": [
+    {
+      "tag": "sitecore-xp1-custom-spe-mssql-init:10.0.0-${windowsservercore_version}",
+      "build-options": [
+        "--memory 4GB",
+        "--build-arg BASE_IMAGE=scr.sitecore.com/sxp/sitecore-xp1-mssql-init:10.0.0-${windowsservercore_version}",
+        "--build-arg MODULE_ASSETS=scr.sitecore.com/sxp/modules/spe-assets:6.1.1-1809",
+        "--build-arg MODULE_NAME=spe"
+      ]
+    },
+    {
+      "tag": "sitecore-xp1-custom-spe-ps-mssql-init:10.0.0-${windowsservercore_version}",
```

Comment:
@jeanfrancoislarente Should this be removed as well? It is still causing the build to fail. Same for any other image that uses the `"--build-arg MODULE_ASSETS=modules/custom-ps-assets:10.0.0-1809"` asset image.

Reply:
Can you share some of the calls you're making? I'm not getting any (more) failures. I'm pruning now to do a full rebuild

Reply:
This is the build command that I am using:

```
.\Build.ps1 -SitecoreUsername "username" -SitecorePassword "password" -SitecoreVersion 10.0.0 -Topology "xp" -OSVersion ltsc2019  -IncludeExperimental -IncludeSpe -IncludeSxa -IncludeJss
```

Reply:
I've added the experimental setting on anything PS...

Can you confirm that the custom-ps-assets is building correctly?

Reply:
Yes, without the `-IncludeExperimental` parameter the build process succeed. It is failing when including the experimental images.

The `custom-ps-assets` image is built successfully, but then the first one that refers to it in the build arguments fails:

```
13:47:55:012: Processing 12 of 24 'sitecore-custom-ps:10.0.0-ltsc2019'...
Sending build context to Docker daemon  6.144kB
Step 1/19 : ARG BASE_IMAGE
Step 2/19 : ARG ASSETS_IMAGE
Step 3/19 : FROM $ASSETS_IMAGE as assets
 ---> 60a814464e4b
Step 4/19 : FROM ${BASE_IMAGE} as build
 ---> 987b1d5e0abf
Step 5/19 : ARG ASSETS_USE_PS_WDP
 ---> Using cache
 ---> 1faef91bb2f6
Step 6/19 : ARG CONN_STRING_CORE
 ---> Using cache
 ---> 64b7bed19f84
Step 7/19 : ARG CONN_STRING_MASTER
 ---> Using cache
 ---> 8f429956add5
Step 8/19 : ARG CONN_STRING_WEB
 ---> Using cache
 ---> 94b2e538aaeb
Step 9/19 : SHELL ["powershell", "-NoProfile", "-Command", "$ErrorActionPreference = 'Stop';"]
 ---> Using cache
 ---> f10e3b31b323
Step 10/19 : WORKDIR /app/
 ---> Using cache
 ---> 73bed344ee89
Step 11/19 : COPY --from=assets ["${ASSETS_USE_PS_WDP}", "/packages/"]
 ---> Using cache
 ---> 3be856248b36
Step 12/19 : COPY scripts/ .
 ---> Using cache
 ---> 7e15f0fe5d2b
Step 13/19 : RUN Expand-Archive -Path $env:ASSETS_USE_PS_WDP -DestinationPath "/app"
 ---> Using cache
 ---> 658ac05b8778
Step 14/19 : FROM ${BASE_IMAGE} as final
 ---> 987b1d5e0abf
Step 15/19 : WORKDIR /app/
 ---> Using cache
 ---> 7f8983016505
Step 16/19 : COPY --from=build /app .
 ---> Using cache
 ---> e89f8f2f19ae
Step 17/19 : ENV ASPNETCORE_URLS=http://+:80     DOTNET_RUNNING_IN_CONTAINER=true     DOTNET_RUNNING_IN_CONTAINERS=true
 ---> Using cache
 ---> f6b1268eedd2
Step 18/19 : EXPOSE 80
 ---> Using cache
 ---> 74efe140bb70
Step 19/19 : ENTRYPOINT [ "powershell", "/app/start.ps1" ]
 ---> Using cache
 ---> 398bb37d6482
Successfully built 398bb37d6482
Successfully tagged sitecore-custom-ps:10.0.0-ltsc2019
13:48:06:942: Build completed for sitecore-custom-ps:10.0.0-ltsc2019. Time: 00:00:11.928.
13:48:06:944: Done with 'sitecore-custom-ps:10.0.0-ltsc2019', but not pushed since 'Registry' was empty.
13:48:06:945: Processing 13 of 24 'sitecore-xp1-custom-spe-ps-mssql-init:10.0.0-ltsc2019'...
Sending build context to Docker daemon  9.216kB
Step 1/7 : ARG BASE_IMAGE
Step 2/7 : ARG MODULE_ASSETS
Step 3/7 : FROM ${MODULE_ASSETS} as module_assets
 ---> 2c7e23f3f20f
Step 4/7 : FROM $BASE_IMAGE as final
 ---> 7055fc3995c6
Step 5/7 : ARG MODULE_NAME
 ---> Using cache
 ---> 109a4577446c
Step 6/7 : SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
 ---> Using cache
 ---> 429b38f973d4
Step 7/7 : COPY --from=module_assets C:\module\db c:\resources\$MODULE_NAME
COPY failed: CreateFile \\?\Volume{09c69779-909c-47c2-a4f8-fd4d38e25580}\module\db: The system cannot find the file specified.
Failed: docker image build --isolation 'hyperv' --memory 4GB --build-arg
BASE_IMAGE=sitecore-xp1-custom-spe-mssql-init:10.0.0-ltsc2019 --build-arg
MODULE_ASSETS=modules/custom-ps-assets:10.0.0-1809 --build-arg MODULE_NAME=ps --tag
'sitecore-xp1-custom-spe-ps-mssql-init:10.0.0-ltsc2019'
'C:\Docker\docker-images\windows\10.x.x\sitecore-custom-mssql\sitecore-custom-modules-mssql-init'
At C:\Docker\docker-images\modules\SitecoreImageBuilder\1.0.0\Public\Invoke-Build.ps1:234 char:74
+ ... here-Object { $_ } | ForEach-Object { throw "Failed: $buildCommand" }
+                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Failed: docker ...les-mssql-init':String) [], RuntimeException
    + FullyQualifiedErrorId : Failed: docker image build --isolation 'hyperv' --memory 4GB --build-arg BASE_IMAGE=site
   core-xp1-custom-spe-mssql-init:10.0.0-ltsc2019 --build-arg MODULE_ASSETS=modules/custom-ps-assets:10.0.0-1809 --bu
  ild-arg MODULE_NAME=ps --tag 'sitecore-xp1-custom-spe-ps-mssql-init:10.0.0-ltsc2019' 'C:\Docker\docker-images\wind
 ows\10.x.x\sitecore-custom-mssql\sitecore-custom-modules-mssql-init'
```

It seems that there is no `C:\module\db` path in the assets image.

Reply:
So there may be something wrong with your converted Publishing Service Module file (scwdp)

Make sure you clean out old .zip files from the build context of ps-assets, try to rebuild it

Then to test:

```powershell
docker run -it modules/custom-ps-assets:10.0.0-1809 cmd
dir module\db
```
Should see: `Sitecore.core.dacpac`

Reply:
Just following up, the scwdp.zip is likely your culprit as I rebuilt everything from scratch and didn't run into any errors.

Reply:
> Make sure you clean out old .zip files from the build context of ps-assets, try to rebuild it

That was it! Thank you! The build process wasn't replacing the old wrong zip file in the `\windows\10.0.0\modules\ps-assets` build context, even if I had the correctly generated scwdp.zip file in the `packages` folder.

All images are building successfully now.

Reply:
Great to hear, glad you got it all figured out!

`git clean -xdf` is your friend (once you've confirmed you don't have important untracked files) :)